netebpfext/net_ebpf_ext_bind.c

```diff
@@ -29,12 +22,96 @@ static ebpf_program_data_t _ebpf_bind_program_data = {&_ebpf_bind_program_info,
 static ebpf_extension_data_t _ebpf_bind_program_info_provider_data = {
     NET_EBPF_EXTENSION_NPI_PROVIDER_VERSION, sizeof(_ebpf_bind_program_data), &_ebpf_bind_program_data};
 
-// b9707e04-8127-4c72-833e-05b1fb439496
-DEFINE_GUID(EBPF_ATTACH_TYPE_BIND, 0xb9707e04, 0x8127, 0x4c72, 0x83, 0x3e, 0x05, 0xb1, 0xfb, 0x43, 0x94, 0x96);
+// Net eBPF Extension Bind Program Information NPI Provider Module GUID: 6c8d3dbd-f1e3-4c42-abb8-cf7f095c9df3
+const NPI_MODULEID DECLSPEC_SELECTANY _ebpf_bind_program_info_provider_moduleid = {
+    sizeof(NPI_MODULEID), MIT_GUID, {0x6c8d3dbd, 0xf1e3, 0x4c42, {0xab, 0xb8, 0xcf, 0x7f, 0x09, 0x5c, 0x9d, 0xf3}}};
 
 // 608c517c-6c52-0x4a26-b677-bb01c34425adf
 DEFINE_GUID(EBPF_PROGRAM_TYPE_BIND, 0x608c517c, 0x6c52, 0x4a26, 0xb6, 0x77, 0xbb, 0x1c, 0x34, 0x42, 0x5a, 0xdf);
 
+const NPI_PROVIDER_CHARACTERISTICS _ebpf_bind_program_info_provider_characteristics = {
+    0,
+    sizeof(NPI_PROVIDER_CHARACTERISTICS),
+    net_ebpf_extension_program_info_provider_attach_client,
+    net_ebpf_extension_program_info_provider_detach_client,
+    NULL,
+    {0,
+     sizeof(NPI_REGISTRATION_INSTANCE),
+     &EBPF_PROGRAM_TYPE_BIND,
+     &_ebpf_bind_program_info_provider_moduleid,
+     0,
+     &_ebpf_bind_program_info_provider_data},
+};
+
+static net_ebpf_extension_program_info_provider_t* _ebpf_bind_program_info_provider_context = NULL;
+
+//
+// Bind Hook NPI Provider.
+//
+ebpf_attach_provider_data_t _net_ebpf_bind_hook_provider_data;
+
+ebpf_extension_data_t _net_ebpf_extension_bind_hook_provider_data = {
+    EBPF_ATTACH_PROVIDER_DATA_VERSION, sizeof(_net_ebpf_bind_hook_provider_data), &_net_ebpf_bind_hook_provider_data};
+
+// Net eBPF Extension Bind Hook NPI Provider Module GUID: eab8f3d9-ab6c-422e-994c-7a80943bc920
+const NPI_MODULEID DECLSPEC_SELECTANY _ebpf_bind_hook_provider_moduleid = {
+    sizeof(NPI_MODULEID), MIT_GUID, {0xeab8f3d9, 0xab6c, 0x422e, {0x99, 0x4c, 0x7a, 0x80, 0x94, 0x3b, 0xc9, 0x20}}};
+
+// b9707e04-8127-4c72-833e-05b1fb439496
+DEFINE_GUID(EBPF_ATTACH_TYPE_BIND, 0xb9707e04, 0x8127, 0x4c72, 0x83, 0x3e, 0x05, 0xb1, 0xfb, 0x43, 0x94, 0x96);
```

Comment:
This same GUID is in ebpf_api.h.   Can we move the literal value to a common header instead of hard coding it in two places?   My fear is that if we change it in one place and not the other, things will break.   It's better to only have the actual value be in one file.

Reply:
I have moved these GUIDs to a new common header. But, ideally, EC should not have any reference to any specific program or attach types. Currently libbpf refers to these pending fix to #223. Also EC refers to this until we come up with a way to verify a program independent of having extensions installed on dev box.